syntax = "proto3";

//protoc --grpc_out=. --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` search.proto
//protoc --cpp_out=. search.proto
package pb;

service Facedb {
    rpc Search (SearchRequest) returns (SearchReply) {}
    rpc GroupList(SearchRequest) returns (GroupsReply) {}
    rpc AddUser(UserInfo) returns (NomalReply) {}
    rpc DelUser(UserInfo) returns (NomalReply) {}
}

message UserInfo{
    string id = 1;
    repeated float feature = 2;
}

message NomalReply{
    bool ret = 1;
}

message GroupsReply{
    repeated string grops = 1;
}

message SearchRequest {
    repeated  float name = 1;
}

message SearchReply {
    message User {
        string name = 1;
        float  distance = 2;
    }
    repeated User users  = 1;
}